spring/boot-autoconfigure/src/test/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtilTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.spring;
+
+import static com.linecorp.armeria.internal.spring.ArmeriaConfigurationUtil.configureAnnotatedHttpServices;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.util.function.Function;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.spring.AnnotatedServiceRegistrationBean;
+import com.linecorp.armeria.spring.MeterIdPrefixFunctionFactory;
+
+public class ArmeriaConfigurationUtilTest {
+
+    @Test
+    public void makesSureDecoratorsAreConfigured() {
+        final Function<Service<HttpRequest, HttpResponse>,
+                ? extends Service<HttpRequest, HttpResponse>> decorator = spy(new DecoratorFunction());
+        final AnnotatedServiceRegistrationBean bean = new AnnotatedServiceRegistrationBean()
+                .setServiceName("test")
+                .setService(new SimpleService())
+                .setDecorators(decorator);
+
+        configureAnnotatedHttpServices(new ServerBuilder(), ImmutableList.of(bean),
+                                       MeterIdPrefixFunctionFactory.DEFAULT);
+        verify(decorator).apply(any());
+
+        reset(decorator);
+
+        configureAnnotatedHttpServices(new ServerBuilder(), ImmutableList.of(bean), null);
+        verify(decorator).apply(any());
+    }
+
+    /**
+     * A decorator function which is the same as {@link #identity()} but is not a final class.
+     */
+    static class DecoratorFunction
+            implements Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> {
+        @Override
+        public Service<HttpRequest, HttpResponse> apply(Service<HttpRequest, HttpResponse> delegate) {
+            return delegate;
```

Comment:
Could you also add the test case where a decorator function actually returns a decorated instance?

Reply:
Thanks! Added.